packager/packager.cc

```diff
@@ -248,15 +274,31 @@ ast::ExpressionPtr prependName(ast::ExpressionPtr scope, core::NameRef name) {
     // If `lastConstLit` is `nullptr`, then `scope` should be EmptyTree.
     ENFORCE(lastConstLit != nullptr || ast::isa_tree<ast::EmptyTree>(scope));
 
-    auto scopeToPrepend = name2Expr(name, name2Expr(core::Names::Constants::PackageRegistry()));
     if (lastConstLit == nullptr) {
-        return scopeToPrepend;
+        return toPrepend;
     } else {
-        lastConstLit->scope = move(scopeToPrepend);
+        lastConstLit->scope = move(toPrepend);
         return scope;
     }
 }
 
+// Prefix a constant reference with a name: `Foo::Bar` -> `<REGISTRY>::<name>::Foo::Bar`
+// Registry is either <PackageRegistry> or <PackageTests>. The latter if following the convention
+// that if scope starts with `Test::`.
+ast::ExpressionPtr prependPackageScope(ast::ExpressionPtr scope, core::NameRef mangledName) {
+    auto *lastConstLit = ast::cast_tree<ast::UnresolvedConstantLit>(scope);
+    ENFORCE(lastConstLit != nullptr);
```

Comment:
Nit: `cast_tree_nonnull` makes this `ENFORCE` for you in debug builds, and is slightly faster than `cast_tree` in release builds because it doesn't check if the cast failed to return `null`.

If does make the LHS a reference, though, so some lines below need to be changed.

```suggestion
    auto lastConstLit = ast::cast_tree_nonnull<ast::UnresolvedConstantLit>(scope);
```

Reply:
Maybe I'm missing something but wasn't sure how to get that working properly with the reassignment in the loop below. 

I'll switch it to ` auto *lastConstLit = &ast::cast_tree_nonnull<ast::UnresolvedConstantLit>(scope);`